### PR TITLE
Add Ripple submitter

### PR DESCRIPTION
**Pull request by servantftechnicolor**
Add a submitter for an in-house tool called ripple to compute on renderfarm.

### CODE REVIEW

File: meshroom/submitters/rippleSubmitter.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            toRemove = []
+
+            #Loop over all nodes in the graph
+            for node in tree.keys():
+
+                #Ignore a node already processed
+                found = False
+                for nodeDone in nodesDone:
+                    if nodeDone == node.name:
+                        found = True
```
Comment (fabiencastan) [inline meshroom/submitters/rippleSubmitter.py:108]:
```suggestion
                if node.name in nodesDone:
                    found = True
```

File: meshroom/submitters/rippleSubmitter.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        waitsFor = []
+        for parent in parents:
+            waitsFor.append(parent.name)
+
+        #Basic command line for this node
+        command='meshroom_compute --node {nodeName} "{meshroomFile}" --extern'.format(nodeName=node.name, meshroomFile=meshroomFile)
+
+        if node.isParallelized:
+            _, _, nbBlocks = node.nodeDesc.parallelization.getSizes(node)
+
```
Comment (fabiencastan) [inline meshroom/submitters/rippleSubmitter.py:60]: maxProcessors = (Level.INTENSIVE in (node.nodeDesc.ram, node.nodeDesc.cpu)) ? 4 : 1

**Comment by natowi** [comment]
It would be great if you could provide a short comment on ripple :)
Is it https://github.com/robclu/ripple ?

**Comment by fabiencastan** [comment]
Hi @natowi ,
It's a internal lib that we use for renderfarm that has no added value externally.
I updated the PR description.

File: meshroom/submitters/rippleSubmitter.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                if found:
+                    continue
+
+                #Check if all parents are already visited
+                valid = True
+                for parent in tree[node]:
+                    found = False
+                    for nodeDone in nodesDone:
+                        if nodeDone == parent.name:
```
Comment (fabiencastan) [inline meshroom/submitters/rippleSubmitter.py:118]: same here

File: meshroom/submitters/rippleSubmitter.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        #Build a tree
+        tree = {}
+        for node in nodes:
+            tree[node] = []
+
+        for end, start in edges:
+            tree[end].append(start)
+
+        nodesDone = []
```
[the remark this answers is not in the document]
Comment (servantftechnicolor) [reply meshroom/submitters/rippleSubmitter.py]: Done

File: meshroom/submitters/rippleSubmitter.py (hunk 1 above)
Comment (servantftechnicolor) [reply meshroom/submitters/rippleSubmitter.py]: Done

File: meshroom/submitters/rippleSubmitter.py (hunk 2 above)
Comment (servantftechnicolor) [reply meshroom/submitters/rippleSubmitter.py:72]: Done

File: meshroom/submitters/rippleSubmitter.py (hunk 3 above)
Comment (servantftechnicolor) [reply meshroom/submitters/rippleSubmitter.py]: done

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**